pkg/virt-controller/watch/pod.go

```diff
@@ -87,22 +88,51 @@ func NewPodControllerWithListWatch(vmCache cache.Indexer, recorder record.EventR
 			vmCopy.ObjectMeta.Labels[corev1.NodeNameLabel] = pod.Spec.NodeName
 			vmCopy.Status.NodeName = pod.Spec.NodeName
 			// Update the VM
-			logger := logging.DefaultLogger().Object(vm)
-			if err := restClient.Put().Resource("vms").Body(&vmCopy).Name(vmCopy.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error(); err != nil {
-				logger.Error().Reason(err).Msg("Setting the VM to pending failed.")
-				if e, ok := err.(*errors.StatusError); ok {
-					if e.Status().Reason == metav1.StatusReasonNotFound {
-						// VM does not exist anymore, we don't have to retry
-						return true
-					}
+			logger := logging.DefaultLogger()
+			if err := putVm(vm, restClient, queue); err != nil {
+				logger.V(3).Info().Msg("Enqueuing VM again.")
+				queue.AddRateLimited(key)
+				return true
+			}
+			logger.Info().Msgf("VM successfully scheduled to %s.", vmCopy.Status.NodeName)
+		} else if vm.Status.Phase == corev1.Running {
+			logger := logging.DefaultLogger()
+			obj, err := kubeapi.Scheme.Copy(vm)
+			if err != nil {
+				logger.Error().Reason(err).Msg("could not copy vm object")
+				// FIXME: should the VM be re-enqueued? failure to copy is a pretty big deal
```

Comment:
Not sure myself here. Probably rate limitting a few times and then dismiss the key and forget about it's enqueue history ... (queue.Forget())